modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -54,27 +54,40 @@ def initialize(info = {})
     deregister_options('RPORT')
   end
 
-  def device_type_int_to_text(device_type)
-    types = [
-      'UNSET', 'BEEP', 'CDROM', 'CDROM FILE SYSTEM', 'CONTROLLER', 'DATALINK',
-      'DFS', 'DISK', 'DISK FILE SYSTEM', 'FILE SYSTEM', 'INPORT PORT', 'KEYBOARD',
-      'MAILSLOT', 'MIDI IN', 'MIDI OUT', 'MOUSE', 'UNC PROVIDER', 'NAMED PIPE',
-      'NETWORK', 'NETWORK BROWSER', 'NETWORK FILE SYSTEM', 'NULL', 'PARALLEL PORT',
-      'PHYSICAL NETCARD', 'PRINTER', 'SCANNER', 'SERIAL MOUSE PORT', 'SERIAL PORT',
-      'SCREEN', 'SOUND', 'STREAMS', 'TAPE', 'TAPE FILE SYSTEM', 'TRANSPORT', 'UNKNOWN',
-      'VIDEO', 'VIRTUAL DISK', 'WAVE IN', 'WAVE OUT', '8042 PORT', 'NETWORK REDIRECTOR',
-      'BATTERY', 'BUS EXTENDER', 'MODEM', 'VDM'
-    ]
-
-    types[device_type]
+  Connected = false
+
+  TYPES = [
```

Comment:
Looks like ruby_smb has a very different definition for these values, it might be worth confirming this - but I don't think this list is appropriate now:

https://github.com/rapid7/ruby_smb/blob/fd6d70e3e6697be0c3815d02e216466bbd7d6bda/lib/ruby_smb/smb1/resource_type.rb#L3-L16

Reply:
I think the only possible share types are
```
s_type = ['DISK', 'PRINTER', 'DEVICE', 'IPC', 'SPECIAL', 'TEMPORARY'][types[t]].gsub("\x00", "")
```
The above is taken from `RubySMB::Dcerpc::Srvsvc` in the NetShareEnumAll class.
I've added those in 👍

Reply:
https://github.com/rapid7/ruby_smb/blob/fd6d70e3e6697be0c3815d02e216466bbd7d6bda/lib/ruby_smb/dcerpc/srvsvc/net_share_enum_all.rb#L95 is the line in question for future traveler's reference. Always good to link to the code line in case it changes down the line :)

Reply:
This is now in the code at https://github.com/rapid7/metasploit-framework/pull/15854/files#diff-0c4fbafe564772e0c82ba399de2a929787774fe92b822eb939b10e0e2d4b7eeaR56-R58

@AlanFoster I think this list is now okay but can you just confirm?